accessman_test.go

```diff
@@ -703,3 +703,111 @@ func TestAddPeerAccessOutbound(t *testing.T) {
 	expecedScore = peerSlotStatus{state: peerStatusTemporary}
 	require.Equal(t, expecedScore, score)
 }
+
+// TestRemovePeerAccess asserts `removePeerAccess` correctly update the
+// accessman's internal state based on the peer's status.
+func TestRemovePeerAccess(t *testing.T) {
+	t.Parallel()
+
+	// Create a testing accessMan.
+	a := &accessMan{
+		peerChanInfo: make(map[string]channeldb.ChanCount),
+		peerScores:   make(map[string]peerSlotStatus),
+	}
+
+	// numRestrictedExpected specifies the final value to expect once the
+	// test finishes.
+	var numRestrictedExpected int
```

Comment:
Nit: Maybe start from 1 here so we do not decrement to -1 ?

Reply:
yeah i kinda wanna show that this num can be negative as we need underflow protection.